src/tcgwars/logic/impl/gen4/RisingRivals.groovy

```diff
@@ -491,28 +631,45 @@ public enum RisingRivals implements LogicCardInfo {
         return evolution (this, from:"Lairon", hp:HP130, type:METAL, retreatCost:3) {
           weakness R, PLUS30
           resistance P, MINUS20
+          customAbility {
+            delayed (priority: LAST) {
+              before APPLY_ATTACK_DAMAGES, {
+                if(bg().currentTurn==self.owner.opposite) {
+                  turnCount=bg.turnCount
+                  lastDamage=bg().dm().find({it.to==self && it.dmg.value>=0})?.dmg
+                }
+              }
+            }
+          }
```

Comment:
This really seems like something that should be on the PokemonCardSet somehow and not a customAbility.

Reply:
I agree, but I'm not sure it should be on the PCS as this effect and every other similar effect i've looked at don't allow you to reflect damage done to pre-evolutions

Reply:
I'm not sure I understand. You're not checking for pre-evolutions anywhere here anyway?

Just keeping a variable on the PCS that gets updated with the last damage value actually done to it seems like a much better way to handle this than a customAbility too.

Reply:
See how the static `wasSwitchedOutThisTurn` works, and I'd say try to emulate that.